src/python/pants/backend/experimental/python/lockfile_metadata.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+from dataclasses import dataclass
+
+if TYPE_CHECKING:
+    from pants.backend.experimental.python.lockfile import PythonLockfileRequest
+
+
+BEGIN_LOCKFILE_HEADER = b"# --- BEGIN PANTS LOCKFILE METADATA: DO NOT EDIT OR REMOVE ---"
+END_LOCKFILE_HEADER = b"# --- END PANTS LOCKFILE METADATA ---"
+
+
+@dataclass
+class LockfileMetadata:
+    invalidation_digest: str | None
+
+
+# Lockfile metadata for headers
+def lockfile_content_with_header(req: PythonLockfileRequest, content: bytes) -> bytes:
+    """ Returns a version of the lockfile with a pants metadata header prepended. """
+    return b"%b\n%b" % (lockfile_metadata_header(req.hex_digest), content)
```

Comment:
You can directly take the `hex_digest` in this arg. Then you can remove the circular dep w/ `PythonLockfileRequest`

Reply:
👍

Reply:
@Eric-Arellano Now that the circular dep is gone, how do you feel about moving the contents of `lockfile_metadata` back to `lockfile`?